examples/go/unit/unit_test.go

```diff
@@ -161,7 +163,607 @@ func TestArrayTypeParam(t *testing.T) {
+				"page":          0,
+			}
+
+			resp, _ := json.Marshal(response)
+
+			return &http.Response{
+				Body: ioutil.NopCloser(bytes.NewReader(resp)),
+			}, nil
+		},
+		)
+
+	page1 := testClient.EXPECT().SendRequest(
+		gomock.Any(),
+		gomock.Any(),
+		gomock.Any(),
+		gomock.Any()).
+		DoAndReturn(func(method string, rawURL string, data url.Values,
+			headers map[string]interface{}) (*http.Response, error) {
+			response := map[string]interface{}{
+				"end":            2,
+				"first_page_uri": "/2010-04-01/Accounts/AC12345678123456781234567812345678/Messages.json?From=9999999999&PageNumber=&To=4444444444&PageSize=5&Page=0",
+				"messages": []map[string]interface{}{
+					{
+						"direction": "outbound-api",
+						"from":      "4444444444",
+						"to":        "9999999999",
+						"body":      "Message 5",
+						"status":    "delivered",
+					},
+					{
+						"direction": "outbound-api",
+						"from":      "4444444444",
+						"to":        "9999999999",
+						"body":      "Message 6",
+						"status":    "delivered",
+					},
+					{
+						"direction": "outbound-api",
+						"from":      "4444444444",
+						"to":        "9999999999",
+						"body":      "Message 7",
+						"status":    "delivered",
+					},
+				},
+				"uri":           "“/2010-04-01/Accounts/AC12345678123456781234567812345678/Messages.json?From=9999999999&PageNumber=&To=4444444444&PageSize=5&Page=1&PageToken=",
+				"page_size":     2,
+				"start":         0,
+				"next_page_uri": "/2010-04-01/Accounts/AC12345678123456781234567812345678/Messages.json?From=9999999999&PageNumber=&To=4444444444&PageSize=5&Page=2&PageToken=PASMc49f620580b24424bcfa885b1f741130",
+				"page":          1,
+			}
+
+			resp, _ := json.Marshal(response)
+
+			return &http.Response{
+				Body: ioutil.NopCloser(bytes.NewReader(resp)),
+			}, nil
+		},
+		)
+
+	gomock.InOrder(
+		page0,
+		page1,
+	)
+
+	twilio := openapi.NewApiServiceWithClient(testClient)
+
+	params := &openapi.ListMessageParams{}
+	params.SetFrom("4444444444")
+	params.SetTo("9999999999")
+	params.SetPageSize(5)
+	params.SetLimit(8)
+
+	messageCount := 0
+
+	channel, _ := twilio.StreamMessage(params)
+	for record := range channel {
+		text := fmt.Sprintf("Message %d", messageCount)
+		assert.Equal(t, text, *record.Body)
+		messageCount += 1
+	}
+
+	assert.Equal(t, 8, messageCount)
+}
+
+func TestStreamError(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	testClient := NewMockBaseClient(mockCtrl)
+
+	testClient.EXPECT().AccountSid().DoAndReturn(func() string {
+		return "AC222222222222222222222222222222"
+	}).AnyTimes()
+
+	testClient.EXPECT().SendRequest(
+		gomock.Any(),
+		gomock.Any(),
+		gomock.Any(),
+		gomock.Any()).
+		DoAndReturn(func(method string, rawURL string, data url.Values,
+			headers map[string]interface{}) (*http.Response, error) {
+			response := map[string]interface{}{
```

Comment:
do we need to include the response body in error tests?

Reply:
We don't have to but that test is replicating the scenario where that could happen.